extensions-core/datasketches/src/main/java/org/apache/druid/query/aggregation/datasketches/LittleEndianWrapper.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.query.aggregation.datasketches;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+import com.yahoo.memory.Memory;
+import com.yahoo.memory.WritableMemory;
+
+public class LittleEndianWrapper
+{
+
+  public static Memory wrap(final ByteBuffer buf)
+  {
+    final ByteOrder originalOrder = buf.order();
+    final Memory mem = Memory.wrap(buf.order(ByteOrder.LITTLE_ENDIAN));
```

Comment:
I wonder why Memory.wrap(..) and WritableMemory.wrap(..) don't handle this conversion instead of forcing all systems calling those methods?
also, this is kind of error prone in case we miss calling wrap via LittleEndianWrapper and call Memory.wrap(..) directly in some place.

Reply:
Also, would it be simple to use ` public static Memory wrap(final ByteBuffer byteBuf, final ByteOrder byteOrder)` here?

Reply:
This is about wrapping a ByteBuffer. New version of the memory library handles the byte order of the incoming ByteBuffer correctly (before it was simplistic and forgiving). But Druid sometimes does not set the byte order correctly. So now it is necessary to force the byte order of sketches to be little endian. And it must be done externally to the memory library because it is really about Druid not setting it correctly.

Reply:
> would it be simpler to use public static Memory wrap(final ByteBuffer byteBuf, final ByteOrder byteOrder) here?

This is a good point. I will think about it.

Reply:
> I see some direct calls to Memory.wrap() still remain

They still remain where the input is not a ByteBuffer

Reply:
so, are we saying that sketch objects are "always" serialized in little endian and when druid loads those bytes in the ByteBuffer, the order is big endian (jvm default) and that needs to be corrected and memory doesn't know that it is getting a sketch object and wouldn't be able to correctly handle the contents ?

Reply:
Please add comments to these methods on why the wrapping is necessary and why it's only needed for ByteByffer and not byte[]